### PR TITLE
Move to Typescript

### PR DESCRIPTION
**Pull request by victoriabernard92**
This PR is the first move towards TS. 

This being a small yet depended on repo, it's important it has typing. This will help with a variety of confusions and pitfalls of this repo. 

Inspired by this issue https://github.com/cloudflare/kv-asset-handler/issues/54

In the future will add even more strict typing this is just getting it all set up

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+module.exports = {
+  entry: './src/index.ts',
+  output: {
+    filename: './bundle.js',
+  },
+
+  // Enable sourcemaps for debugging webpack's output.
+  devtool: 'source-map',
```
Comment (EverlastingBugstopper) [inline webpack.config.js]: will this create source maps for production builds? as in will this bloat user's worker scripts?

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "types": "./dist/index.d.ts",
   "scripts": {
     "build": "tsc -d",
-    "build:watch": "yarn build -w",
```
Comment (EverlastingBugstopper) [inline package.json]: can we just make this `tsc -dw` instead so folks can use whatever CLI they want?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "compilerOptions": {
+    "outDir": "./dist",
+    "noImplicitAny": true,
+    "target": "es5",
+    "sourceMap": true,
```
Comment (EverlastingBugstopper) [inline tsconfig.json]: same source map question remains here - i'm wondering if there's a way to configure this so it only generates source maps when developing and not when builds are production

**Comment by victoriabernard92** [comment]
> lgtm. probably worth making an issue for looking into source maps - i imagine they're very useful for developing and that there's a clear path for how to use them without bundling them for prod builds

It works when I remove it. So I just going to remove it for now and think about adding it back later

**Pull request merged**